server_test.go

```diff
@@ -153,15 +153,21 @@ func (s) TestRetryChainedInterceptor(t *testing.T) {
 	handler := func(ctx context.Context, req any) (any, error) {
 		return nil, nil
 	}
-	ii(context.Background(), nil, nil, handler)
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+
+	ii(ctx, nil, nil, handler)
 	if !cmp.Equal(records, []int{1, 2, 3, 2, 3}) {
 		t.Fatalf("retry failed on chained interceptors: %v", records)
 	}
 }
 
 func (s) TestStreamContext(t *testing.T) {
 	expectedStream := &transport.Stream{}
-	ctx := NewContextWithServerTransportStream(context.Background(), expectedStream)
+	ctx, cancel := context.WithTimeout(NewContextWithServerTransportStream(context.Background(), expectedStream), defaultTestTimeout)
```

Comment:
I believe we would want the same change here as you requested in the metadata tests. So it would be more readable.